### PR TITLE
Ensure a request for telemetry happens in Condition Sets

### LINKED ISSUE #7484: Condition sets don't show most recent status for Operator/Mission Status on nav
#### Summary

When navigating to a Condition set based on Mission Status / Operator Status values (both enum values, unconfirmed if this extends to all enum telemetry), the Condition Set will not get the most recent value. However, if one reloads the page while viewing the condition set, or the state is changed while viewing the condition set, it will evaluate properly.

#### Expected vs Current Behavior

Condition sets should get the most recent value on nav to the object

#### Steps to Reproduce

1. Create a Condition Set with conditions based on Operator Status or Mission Status values
2. While viewing the condition set, set a value in Operator/Mission status
3. Notice the value is received by the condition set and it evaluates
4. Nav away to another object and back again
5. Notice the condition set value is '---' and nothing is evaluated
6. Refresh the page
7. Notice the condition set gets the most recent value and it evaluates properly

#### Environment

* Open MCT Version:  0aceb4b590be32d0abfc172b972b82b8c2bbcbfd
* Deployment Type:  testathon
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [x] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information

https://github.com/nasa/openmct/assets/9259993/a6ca545d-a6fa-43a3-b9ee-3c13e5782291

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7484 

### Describe your changes:

When subscribing to telemetry in a Condition Set, we need to ensure we request telemetry in addition to a simple subscribe. This is because on telemetry like operator status, the Telemetry API may already have a subscription (due to the poll widget), and thus not create a new subscription, and thus not immediately fire a callback.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by scottbell** [comment]
The PR in action:

https://github.com/nasa/openmct/assets/9853862/8c776611-c47e-448d-85b9-05e1a37e3431

Note we don't have a great way of writing e2e tests for this right now as our YAMCS setup doesn't have:
1. Users
2. Operator Status Telemetry

File: src/plugins/condition/ConditionManager.js [hunk 1]
```diff
@@ -91,7 +102,7 @@ export default class ConditionManager extends EventEmitter {
 
     //force re-computation of condition set result as we might be in a state where
     // there is no telemetry datum coming in for a while or at all.
-    let latestTimestamp = getLatestTimestamp(
+    const latestTimestamp = getLatestTimestamp(
       {},
       {},
       this.timeSystems,
```
Comment (ozyx) [inline src/plugins/condition/ConditionManager.js:108]: this function call looks insane lol

File: src/plugins/condition/ConditionManager.js [hunk 2]
```diff
@@ -56,20 +56,31 @@ export default class ConditionManager extends EventEmitter {
...
     const metadata = this.openmct.telemetry.getMetadata(endpoint);
 
-    this.telemetryObjects[id] = Object.assign({}, endpoint, {
+    this.telemetryObjects[telemetryKeyString] = Object.assign({}, endpoint, {
       telemetryMetaData: metadata ? metadata.valueMetadatas : []
     });
-    this.subscriptions[id] = this.openmct.telemetry.subscribe(
+
+    // get latest telemetry value (in case subscription is cached and no new data is coming in)
+    this.requestLatestValue(endpoint);
```
Comment (ozyx) [inline src/plugins/condition/ConditionManager.js:81]: that'll fix it i reckon. nice

File: src/plugins/condition/ConditionManager.js [hunk 3]
```diff
@@ -56,20 +56,31 @@ export default class ConditionManager extends EventEmitter {
...
-    const id = this.openmct.objects.makeKeyString(endpoint.identifier);
-    if (this.subscriptions[id]) {
-      console.log('subscription already exists');
+  async requestLatestValue(endpoint) {
+    const options = {
+      size: 1,
+      strategy: 'latest'
+    };
+    const latestData = await this.openmct.telemetry.request(endpoint, options);
+    this.telemetryReceived(endpoint, latestData[0]);
```
Comment (ozyx) [inline src/plugins/condition/ConditionManager.js:65]: should we check to see that latestData is not falsy or empty array first? or do we want it to fail fast here?

**Review by ozyx — COMMENTED** [review]
Just a question

File: src/plugins/condition/ConditionManager.js (hunk 3 above)
Comment (scottbell) [reply src/plugins/condition/ConditionManager.js:65]: heuristically, i'm inclined to let it fail fast here

Comment (akhenry) [reply src/plugins/condition/ConditionManager.js:65]: If a condition is expected then handle it, if it's unexpected fail fast to help developers fix it. An empty array being returned from a telemetry provided is valid and expected. A falsy response is not.

**Review by akhenry — APPROVED** [review]
Looks good. Will file a followup to refactor Condition Sets to use TelemetryCollections which will remove a lot of boilerplate here.

Edit: Filed https://github.com/nasa/openmct/issues/7620

**Comment by scottbell** [comment]
> Looks good. Will file a followup to refactor Condition Sets to use TelemetryCollections which will remove a lot of boilerplate here.
> 
> Edit: Filed #7620

Wonderful! This was my thought too, but seemed ambitious for a bugfix.

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**